consensus/service/src/byzantine_ledger/mod.rs

```diff
@@ -274,57 +274,152 @@ mod tests {
+        peer_ids
+            .iter()
+            .map(|peer_id| {
+                let signer_key = Ed25519Pair::from_random(rng);
+                let uri = test_peer_uri(*peer_id, hex::encode(&signer_key.public_key()));
+                let node_id = test_node_id(uri.clone(), &signer_key);
+                let quorum_set = QuorumSet::empty();
+                PeerConfig::new(node_id, uri, quorum_set, signer_key)
+            })
+            .collect()
+    }
+
     #[test_with_logger]
-    fn test_single_slot_consensus(logger: Logger) {
-        // Set up `local_node`.
-        let trivial_quorum_set = QuorumSet::empty();
+    fn test_is_behind(logger: Logger) {
+        let mut rng: StdRng = SeedableRng::from_seed([216u8; 32]);
 
-        let mut seeded_rng: FixedRng = SeedableRng::from_seed([0u8; 32]);
+        // Other nodes.
+        let peers = get_peers(&[11, 22, 33], &mut rng);
 
-        let node_a_signer_key = Ed25519Pair::from_random(&mut seeded_rng);
-        let node_a_uri = test_peer_uri(22, hex::encode(&node_a_signer_key.public_key()));
-        let node_a = (
-            test_node_id(node_a_uri.clone(), &node_a_signer_key),
-            trivial_quorum_set.clone(),
-        );
+        // Local node.
+        let (local_node_id, _local_node_uri, msg_signer_key) = get_local_node_config(11);
+
+        // Local node's quorum set.
+        let local_quorum_set =
+            QuorumSet::new_with_node_ids(2, vec![peers[0].id.clone(), peers[1].id.clone()]);
+
+        // Local node's Ledger.
+        let mut ledger = create_ledger();
+        let sender = AccountKey::random(&mut rng);
+        let num_blocks = 1;
+        initialize_ledger(&mut ledger, num_blocks, &sender, &mut rng);
 
-        let node_b_signer_key = Ed25519Pair::from_random(&mut seeded_rng);
-        let node_b_uri = test_peer_uri(33, hex::encode(&node_b_signer_key.public_key()));
-        let node_b = (
-            test_node_id(node_b_uri.clone(), &node_b_signer_key),
-            trivial_quorum_set.clone(),
+        // Mock peer_manager
+        let peer_manager = ConnectionManager::new(
+            vec![
+                MockPeerConnection::new(
+                    peers[0].uri.clone(),
+                    local_node_id.clone(),
+                    ledger.clone(),
+                    10,
+                ),
+                MockPeerConnection::new(
+                    peers[1].uri.clone(),
+                    local_node_id.clone(),
+                    ledger.clone(),
+                    10,
+                ),
+            ],
+            logger.clone(),
         );
 
-        let node_c_signer_key = Ed25519Pair::from_random(&mut seeded_rng);
-        let node_c_uri = test_peer_uri(44, hex::encode(&node_c_signer_key.public_key()));
-        let _node_c = (
-            test_node_id(node_c_uri.clone(), &node_c_signer_key),
-            trivial_quorum_set,
+        // Mock tx_manager
+        let tx_manager = Arc::new(MockTxManager::new());
```

Comment:
This is the auto-generated mock for the TxManager trait. It doesn't really do anything here because this test is very simple, but in general it would make it easy to test how ByzantineLedger reacts to a transaction that is not well-formed, not valid, not in the cache, etc.